File: ga/4omini_result_2024_7/result/1968B_cpp_mini/19_1968B.cpp
use std::io;
use std::io::prelude::*;

fn main() {
    let stdin = io::stdin();
    let mut lines = stdin.lock().lines();
    
    let t: usize = lines.next().unwrap().unwrap().parse().unwrap();
    
    for _ in 0..t {
        // Read x and k
        let line = lines.next().unwrap().unwrap();
        let mut iter = line.split_whitespace();
        let _x: usize = iter.next().unwrap().parse().unwrap();
        let _k: usize = iter.next().unwrap().parse().unwrap();
        
        // Read strings a and b
        let a = lines.next().unwrap().unwrap();
        let b = lines.next().unwrap().unwrap();
        
        let mut ans = 0;

        // Determine how many characters of a can be found sequentially in b
        for char_b in b.chars() {
            if ans < a.len() && a.chars().nth(ans) == Some(char_b) {
                ans += 1;
            }
        }
        
        println!("{}", ans);
    }
}

File: ga/4omini_result_2024_7/result/1968B_mini/19_1968B.rs
use std::io;

fn main() {
    let stdin = io::stdin();
    let mut input = String::new();
    
    stdin.read_line(&mut input).unwrap();
    let t: usize = input.trim().parse().unwrap();

    for _ in 0..t {
        input.clear();
        stdin.read_line(&mut input).unwrap();
        let nums: Vec<usize> = input.split_whitespace()
                                     .map(|x| x.parse().unwrap())
                                     .collect();
        let n = nums[0];
        let m = nums[1];
        
        input.clear();
        stdin.read_line(&mut input).unwrap();
        let sn = input.trim();

        input.clear();
        stdin.read_line(&mut input).unwrap();
        let sm = input.trim();

        let mut i = 0;
        let mut j = 0;

        while i < n && j < m {
            if sn.chars().nth(i) == sm.chars().nth(j) {
                i += 1;
            }
            j += 1;
        }

        println!("{}", i);
    }
}